### PR TITLE
[WIP] feat: add capability to serve static content

### PR DESCRIPTION
**Pull request by b-admike**
#### Related issues

- connect to #691 

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,15 +11,19 @@
   "license": "MIT",
   "dependencies": {},
   "devDependencies": {
-    "@commitlint/cli": "^5.2.5",
-    "@commitlint/config-angular": "^5.1.1",
-    "@commitlint/config-lerna-scopes": "^5.2.0",
+    "@commitlint/cli": "^4.3.0",
```
Comment (raymondfeng) [inline package.json]: Why do we downgrade `@commitlint/*`?

File: package.json [hunk 2]
```diff
@@ -11,15 +11,19 @@
...
     "@types/mocha": "^2.2.44",
     "@types/node": "^8.0.56",
     "coveralls": "^3.0.0",
     "cz-conventional-changelog": "^2.1.0",
     "lerna": "^2.5.1",
-    "mocha": "^4.0.0"
+    "mime": "^2.0.3",
+    "mocha": "^4.0.0",
+    "request": "^2.83.0",
+    "request-promise": "^4.2.2"
```
Comment (raymondfeng) [inline package.json]: Why do we add these dependencies to top-level package.json?

**Comment by raymondfeng** [comment]
There are some interesting details that we need to handle for static file serving. Can we somehow leverage existing modules such as the ones from express/koa?

**Comment by bajtos** [comment]
I'd personally like to see either a Controller-based or handler-function-based approach. IMO, our REST transport should provide enough functionality for controller methods to allow them to implement static file serving as a regular method. Many of these features (e.g. the ability to control response status code & content type, send a stream in the response body) are useful and important outside of static-file-serving context too.

**Comment by b-admike** [comment]
I'm going to close this for now since we have moved the related task out of this sprint and I'm not working on it.

**Pull request closed without merging**